From 63a113ad959485a6943acd2217c483554bccaba6 Mon Sep 17 00:00:00 2001
From: Thomas Perrot <thomas.perrot@bootlin.com>
Date: Tue, 9 Jul 2024 11:23:13 +0200
Subject: [PATCH 2/5] Disable unsafe includes

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Thomas Perrot <thomas.perrot@bootlin.com>
---
 Makefile | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/Makefile b/Makefile
index 45b9f478..1d7841a4 100755
--- a/Makefile
+++ b/Makefile
@@ -219,17 +219,6 @@ SDLCFLAGS := $(shell sdl2-config --cflags)
 
 # ----------
 
-# Base include path.
-ifeq ($(YQ2_OSTYPE),Linux)
-INCLUDE := -I/usr/include
-else ifeq ($(YQ2_OSTYPE),FreeBSD)
-INCLUDE := -I/usr/local/include
-else ifeq ($(YQ2_OSTYPE),OpenBSD)
-INCLUDE := -I/usr/local/include
-else ifeq ($(YQ2_OSTYPE),Windows)
-INCLUDE := -I/usr/include
-endif
-
 # ----------
 
 # Local includes for GLAD.
-- 
2.25.1

