From 1bbef354c6535844ad90e3311e7aadd60984ffc0 Mon Sep 17 00:00:00 2001
From: Thomas Perrot <thomas.perrot@bootlin.com>
Date: Tue, 9 Jul 2024 11:27:27 +0200
Subject: [PATCH 4/5] Remove root protections

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Thomas Perrot <thomas.perrot@bootlin.com>
---
 src/backends/unix/main.c | 22 ----------------------
 1 file changed, 22 deletions(-)

diff --git a/src/backends/unix/main.c b/src/backends/unix/main.c
index c763dbad..5ecbb805 100644
--- a/src/backends/unix/main.c
+++ b/src/backends/unix/main.c
@@ -90,28 +90,6 @@ main(int argc, char **argv)
 		}
 	}
 
-	/* Prevent running Quake II as root. Only very mad
-	   minded or stupid people even think about it. :) */
-	if (getuid() == 0)
-	{
-		printf("Quake II shouldn't be run as root! Backing out to save your ass. If\n");
-		printf("you really know what you're doing, edit src/unix/main.c and remove\n");
-		printf("this check. But don't complain if Quake II eats your dog afterwards!\n");
-
-		return 1;
-	}
-
-	// Enforce the real UID to prevent setuid crap
-	if (getuid() != geteuid())
-	{
-		printf("The effective UID is not the real UID! Your binary is probably marked\n");
-		printf("'setuid'. That is not good idea, please fix it :) If you really know\n");
-		printf("what you're doing edit src/unix/main.c and remove this check. Don't\n");
-		printf("complain if Quake II eats your dog afterwards!\n");
-
-		return 1;
-	}
-
 	// enforce C locale
 	setenv("LC_ALL", "C", 1);
 
-- 
2.25.1

