From 99155acff8d79e4ae810f67beb5c8f170af68393 Mon Sep 17 00:00:00 2001
From: Thomas Perrot <thomas.perrot@bootlin.com>
Date: Tue, 9 Jul 2024 11:28:39 +0200
Subject: [PATCH 5/5] Use pkgconfig

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Thomas Perrot <thomas.perrot@bootlin.com>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 0aac1056..a66d227a 100755
--- a/Makefile
+++ b/Makefile
@@ -215,7 +215,7 @@ ZIPCFLAGS += -DNOUNCRYPT
 # ----------
 
 # Extra CFLAGS for SDL.
-SDLCFLAGS := $(shell sdl2-config --cflags)
+SDLCFLAGS := $(shell pkg-config --cflags sdl2)
 
 # ----------
 
@@ -256,7 +256,7 @@ endif
 ifeq ($(YQ2_OSTYPE), Darwin)
 SDLLDFLAGS := -lSDL2
 else # not Darwin
-SDLLDFLAGS := $(shell sdl2-config --libs)
+SDLLDFLAGS := $(shell pkg-config --libs sdl2)
 endif # Darwin
 
 # The renderer libs don't need libSDL2main, libmingw32 or -mwindows.
-- 
2.25.1

